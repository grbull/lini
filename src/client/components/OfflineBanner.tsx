import cn from 'classnames';
import React, { ReactElement } from 'react';

import { useIsOnline } from '../hooks/useIsOnline';

export function OfflineBanner(): ReactElement {
  const isOnline = useIsOnline();

  if (isOnline) {
    return <div />;
  }

  return (
    <div
      className={cn(
        'fixed',
        'bottom-14 md:bottom-8',
        'h-6',
        'w-full',
        'bg-red-500',
        'flex',
        'items-center',
        'justify-center'
      )}
    >
      You are currently offline.
    </div>
  );
}
