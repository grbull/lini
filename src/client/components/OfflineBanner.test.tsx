/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { OfflineBanner } from './OfflineBanner';

function setup(): RenderResult {
  const utils = render(<OfflineBanner />);

  return { ...utils };
}

describe('OfflineBanner Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot when offline', () => {
    window.addEventListener = jest
      .fn()
      .mockImplementation((event: string, callback) => {
        if (event === 'offline') {
          callback();
        }
      });

    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
