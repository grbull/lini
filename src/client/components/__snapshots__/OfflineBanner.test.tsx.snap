// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfflineBanner Component matches the snapshot 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;

exports[`OfflineBanner Component matches the snapshot when offline 1`] = `
<DocumentFragment>
  <div
    class="fixed bottom-14 md:bottom-8 h-6 w-full bg-red-500 flex items-center justify-center"
  >
    You are currently offline.
  </div>
</DocumentFragment>
`;
